src/Client.php

```diff
@@ -47,7 +47,7 @@ class AvaTaxClientBase
      * @param string $environment  Indicates which server to use; acceptable values are "sandbox" or "production", or the full URL of your AvaTax instance.
      * @param array $guzzleParams  Extra parameters to pass to the guzzle HTTP client (http://docs.guzzlephp.org/en/latest/request-options.html)
      */
-    public function __construct($appName, $appVersion, $machineName, $environment, $guzzleParams = [])
+    public function __construct($appName="", $appVersion="", $machineName="", $environment, $guzzleParams = [])
```

Comment:
AppName and AppVersion should be mandatory.  We should throw an exception if the user provides a null or empty app name / app version.  Machine name should be nullable.

Reply:
The PR has been updated with that change